internal/httphandler/db_request_handler_test.go

```diff
@@ -212,6 +212,187 @@ func TestDBRequestHandler_DBStatus(t *testing.T) {
 	}
 }
 
+func TestDBRequestHandler_DBIndex(t *testing.T) {
+	submittingUserName := "alice"
+	dbName := "testDBName"
+
+	cryptoDir := testutils.GenerateTestClientCrypto(t, []string{"alice", "bob"})
+	aliceCert, aliceSigner := testutils.LoadTestClientCrypto(t, cryptoDir, "alice")
+	_, bobSigner := testutils.LoadTestClientCrypto(t, cryptoDir, "bob")
+
+	testCases := []struct {
+		name               string
+		requestFactory     func() (*http.Request, error)
+		dbMockFactory      func(response *types.GetDBIndexResponseEnvelope) bcdb.DB
+		expectedResponse   *types.GetDBIndexResponseEnvelope
+		expectedStatusCode int
+		expectedErr        string
+	}{
+		{
+			name: "valid dbIndex request",
+			requestFactory: func() (*http.Request, error) {
+				req, err := http.NewRequest(http.MethodGet, constants.URLForGetDBIndex(dbName), nil)
+				if err != nil {
+					return nil, err
+				}
+				req.Header.Set(constants.UserHeader, submittingUserName)
+				sig := testutils.SignatureFromQuery(t, aliceSigner, &types.GetDBIndexQuery{UserId: submittingUserName, DbName: dbName})
+				req.Header.Set(constants.SignatureHeader, base64.StdEncoding.EncodeToString(sig))
+				return req, nil
+			},
+			dbMockFactory: func(response *types.GetDBIndexResponseEnvelope) bcdb.DB {
+				db := &mocks.DB{}
+				db.On("GetCertificate", submittingUserName).Return(aliceCert, nil)
+				db.On("GetDBIndex", dbName, submittingUserName).Return(response, nil)
+				return db
+			},
+			expectedResponse:   &types.GetDBIndexResponseEnvelope{Response: &types.GetDBIndexResponse{Header: &types.ResponseHeader{NodeId: "testNodeID"}}, Signature: []byte{0, 0, 0}},
```

Comment:
where is the `index` itself? fill in something...

Reply:
Done